### PR TITLE
RDCC-2029 Update/Suspend user flow

### PR DESCRIPTION
**Pull request by shreedhar-hmcts**
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/RDCC-2029

### Change description ###
Update/Suspend CW user flow

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[X] No
```

### CODE REVIEW

File: src/functionalTest/java/uk/gov/hmcts/reform/cwrdapi/AuthorizationFunctionalTest.java [hunk 1]
```diff
@@ -58,14 +65,12 @@
     protected static String s2sToken;
 
     public static final String EMAIL = "EMAIL";
-
     public static final String CREDS = "CREDS";
-
-    public static final String BEARER = "BEARER";
-
     public static final String EMAIL_TEMPLATE = "CWR-func-test-user-%s@cwrfunctestuser.com";
```
Comment (nitishsingla1234) [inline src/functionalTest/java/uk/gov/hmcts/reform/cwrdapi/AuthorizationFunctionalTest.java:69]: do we need to change the domain to @justice.gov.uk?

File: src/functionalTest/java/uk/gov/hmcts/reform/cwrdapi/CaseWorkerRefCreateTest.java [hunk 2]
```diff
@@ -84,9 +81,97 @@ public void whenUserNotExistsInCwrAndSidamAndUp_Ac1_LD_disabled() {
...
+    @Test
+    public void whenUserNotExistsInCwrAndUpAndExistsInSidam_Ac2() {
+        List<CaseWorkersProfileCreationRequest> profileCreateRequests = createNewActiveCaseWorkerProfile();
+        UserProfileResponse upResponse = getUserProfileFromUp(profileCreateRequests.get(0).getEmailId());
+        assertEquals(ImmutableList.of(CWD_USER, CASEWORKER_IAC_BULKSCAN), upResponse.getRoles());
+    }
+
+    @Test
+    public void whenUserExistsInCwrAndUpAndExistsInSidam_Ac3() {
+
```
Comment (nitishsingla1234) [inline src/functionalTest/java/uk/gov/hmcts/reform/cwrdapi/CaseWorkerRefCreateTest.java:93]: Just a suggestion, if we can put a comment like "//this test verifies that profiles are created in UP". 
It's quite difficult to grasp by "_Ac2,3,4" etc.

**Review by nitishsingla1234 — APPROVED** [review]

**Comment by nitishsingla1234** [comment]
There are minor code smells.

File: src/functionalTest/java/uk/gov/hmcts/reform/cwrdapi/idam/IdamOpenIdClient.java [hunk 3]
```diff
@@ -86,12 +86,21 @@ public IdamOpenIdClient(TestConfigProperties testConfig) {
         return userCreds;
     }
 
-    public String getInternalOpenIdToken(String role) {
-        if (internalOpenIdTokenPrdAdmin == null || StringUtils.isNotEmpty(role)) {
-            Map<String, String> userCreds = StringUtils.isNotEmpty(role) ? createUser(role) : createUser("cwd-admin");
-            internalOpenIdTokenPrdAdmin = getOpenIdToken(userCreds.get(EMAIL), userCreds.get(CREDS));
+    public String getcwdAdminOpenIdToken() {
+        if (crdAdminToken == null) {
```
Comment (abhijitdiwan) [inline src/functionalTest/java/uk/gov/hmcts/reform/cwrdapi/idam/IdamOpenIdClient.java:90]: use isEmpty or isNull rather hard null

File: src/functionalTest/java/uk/gov/hmcts/reform/cwrdapi/client/CaseWorkerApiClient.java [hunk 4]
```diff
@@ -147,9 +116,70 @@ public RequestSpecification getMultipleAuthHeaders(String userToken) {
...
+                        .roles(roleRequests)
+                        .workerWorkAreaRequests(areaRequests).build());
     }
 
+    public Response createUserProfiles(List<CaseWorkersProfileCreationRequest> caseWorkersProfileCreationRequests) {
+        Response response = getMultipleAuthHeadersInternal()
+                .body(caseWorkersProfileCreationRequests)
+                .post("/refdata/case-worker/users/")
+                .andReturn();
+        log.info(":: Create user profile response status code :: " + response.statusCode());
```
Comment (abhijitdiwan) [inline src/functionalTest/java/uk/gov/hmcts/reform/cwrdapi/client/CaseWorkerApiClient.java:177]: do we need logs in func test cases?

Comment (shreedhar-hmcts) [reply src/functionalTest/java/uk/gov/hmcts/reform/cwrdapi/client/CaseWorkerApiClient.java:177]: just to see what was the response status code. This is useful whenever the functional test cases fails on AAT

File: src/functionalTest/java/uk/gov/hmcts/reform/cwrdapi/idam/IdamOpenIdClient.java (hunk 3 above)
Comment (shreedhar-hmcts) [reply src/functionalTest/java/uk/gov/hmcts/reform/cwrdapi/idam/IdamOpenIdClient.java:90]: yes used isNull

File: src/functionalTest/java/uk/gov/hmcts/reform/cwrdapi/CaseWorkerRefCreateTest.java (hunk 2 above)
Comment (shreedhar-hmcts) [reply src/functionalTest/java/uk/gov/hmcts/reform/cwrdapi/CaseWorkerRefCreateTest.java:93]: put some comments

File: src/functionalTest/java/uk/gov/hmcts/reform/cwrdapi/AuthorizationFunctionalTest.java (hunk 1 above)
Comment (shreedhar-hmcts) [reply src/functionalTest/java/uk/gov/hmcts/reform/cwrdapi/AuthorizationFunctionalTest.java:69]: good point, changed

**Review by nitishsingla1234 — APPROVED** [review]

**Review by SujeethKumarAri — APPROVED** [review]

**Review by abhijitdiwan — APPROVED** [review]

**Review by nitishsingla1234 — APPROVED** [review]

**Review by SujeethKumarAri — APPROVED** [review]

**Review by nitishsingla1234 — APPROVED** [review]

**Pull request merged**